src/dbnode/encoding/m3tsz/iterator_test.go

```diff
@@ -363,3 +364,18 @@ func TestReaderIteratorNextWithUnexpectedTimeUnit(t *testing.T) {
 	require.False(t, it.Next())
 	require.Error(t, it.Err())
 }
+
+func TestReaderIteratorDecodingRegression(t *testing.T) {
+	b64 := "FnLaQ5ZggACAIEEAUAgIAAAJ1XOXnQH+QAAAAAAAAAB4AAOpgUJ3igWXnNAAYAAAAAEuYPEbxWsXrvGZl9ZGm8J3+1YS3vGZjVZdm8RvK6xHuz1rxmZU2R6u8j/a+wE2rxmY02e2vEbzbsRuvGZtzZOq8J3ivYO6vIXOm8ZmL1lq7xG89rEWbxmfVWT/vAf9qrBFm8Zme1jq7yFxevEbzpsW6vGZivZOq8J39zYRuvGZm3Zba8RvGmxNqzr7xmY9WSpu+h+61gR7vGZldZ9m8RvGqxLe8Zm7VkabwneX1g+u8hcVrxmZc2XqvEbxfsSqvGZ/zZFa8B/yrsF/rxmZU2Ras/e8RvKqxFm8Zm91ku7wneK1h+m8ZmXVkb7xG92rEqbxmY7WPbvIXK67if/HmwBaq8Zma9kerxG/qbE3rxmY12W2vCd5s2EarO/vGZnVZFm8RvbaxOu8ZmL1nqbwH/PqwRXvGZtVZPmzrrxG8W7E9rxmbU2RavCd532HurxmY02TWvEb27sRuvGZmzY6q8hcd7vofm1YFqbxmY/WSrvEb/WsR5vGZlVZd+8J3jVYS5s9a8ZmNdkvrxG96bEWrxmZb2fqvAf8WbBK68Zm/dkVrxG8qbD+ryFyvvGZi1Zem8J3ltYRbvGZ/dgA"
+
+	data, err := base64.StdEncoding.DecodeString(b64)
+	require.NoError(t, err)
+
+	it := NewReaderIterator(xio.NewBytesReader64(data), true, encoding.NewOptions())
+	for i := 0; it.Next(); i++ {
+		require.NoError(t, it.Err())
+		it.Current()
+	}
```

Comment:
Thanks for the repro and tracking down the issue!
Can you also validate datapoint-by-datapoint w/ epsilon, too, and log which datapoint fails/has a mismatch?